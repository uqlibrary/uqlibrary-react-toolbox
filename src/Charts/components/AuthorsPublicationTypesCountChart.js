import React from 'react';
import {PropTypes} from 'prop-types';
import Chart from './Chart';

class AuthorsPublicationTypesCountChart extends React.Component {
    static propTypes = {
        className: PropTypes.string,
        series: PropTypes.array
    };

    static defaultProps = {
        series: [{
            name: 'Publications count by type',
            data: [
                ['Journal articles', 200],
                ['Conference papers', 150],
                ['Book chapters', 106],
                ['Books', 30],
                ['Other', 5]
            ]
        }]
    };

    constructor(props) {
        super(props);

        this.state = {
            options: {
                chart: {
                    height: 320,
                    plotShadow: false,
                    plotBorderWidth: 0,
                    spacingBottom: 10,
                    type: 'pie'
                },
                credits: {
                    enabled: false
                },
                legend: {
                    itemStyle: {
                        width: 110
                    },
                    align: 'center',
                    symbolRadius: 0,
                    floating: true,
                    layout: 'vertical',
                    y: -110
                },
                tooltip: {
                    enabled: false
                },
                title: {
                    text: ''
                },
                subtitle: {
                    text: ''
                },
                plotOptions: {
                    pie: {
                        // animation: false,
                        showInLegend: true,
                        startAngle: 45,
                        dataLabels: {
                            style: {
                                color: '#FFFFFF',
                                textOutline: 'none'
                            },
                            allowOverlap: false,
                            distance: 10,
                            padding: 2,
                            softConnector: true,
                            className: 'pieLabels ',
                            format: '{y}',
                            useHTML: false,
                            enabled: true
                        },
                        shadow: false,
                        center: ['50%', '50%'],
                        size: '90%',
                        innerSize: '65%',
                        borderColor: 'none'
                    }
                },
                series: this.props.series
            }
        };
    }

    render() {
        return (
            <Chart className={`${this.props.className || ''} authors-publication-types-count-chart`} chartOptions={this.state.options} />
        );
    }
}

export default AuthorsPublicationTypesCountChart;
