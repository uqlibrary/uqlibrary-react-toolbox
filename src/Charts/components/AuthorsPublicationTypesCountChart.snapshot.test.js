jest.dontMock('./AuthorsPublicationTypesCountChart');

import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import AuthorsPublicationTypesCountChart from './AuthorsPublicationTypesCountChart';

function setup({series, categories, yAxisTitle}) {
    const props = {
        series
    };
    return shallow(<AuthorsPublicationTypesCountChart {...props} />);
}


describe('AuthorsPublicationTypesCountChart snapshot tests', () => {
    it('it should render chart component', () => {
        const app = setup({series: []});
        expect(toJson(app)).toMatchSnapshot();
    });
});

