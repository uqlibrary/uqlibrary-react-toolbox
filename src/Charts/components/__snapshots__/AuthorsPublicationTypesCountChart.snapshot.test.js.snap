// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsPublicationTypesCountChart snapshot tests it should render chart component 1`] = `
<Chart
  chartOptions={
    Object {
      "chart": Object {
        "height": 320,
        "plotBorderWidth": 0,
        "plotShadow": false,
        "spacingBottom": 10,
        "type": "pie",
      },
      "credits": Object {
        "enabled": false,
      },
      "legend": Object {
        "align": "center",
        "floating": true,
        "layout": "vertical",
        "symbolRadius": 0,
        "y": -110,
      },
      "plotOptions": Object {
        "pie": Object {
          "borderColor": "none",
          "center": Array [
            "50%",
            "50%",
          ],
          "dataLabels": Object {
            "allowOverlap": false,
            "className": "pieLabels ",
            "distance": 12,
            "enabled": true,
            "format": "{y}",
            "useHTML": true,
          },
          "innerSize": "65%",
          "shadow": false,
          "showInLegend": true,
          "size": "95%",
          "startAngle": 90,
        },
      },
      "series": Array [],
      "subtitle": Object {
        "text": "",
      },
      "title": Object {
        "text": "",
      },
      "tooltip": Object {
        "enabled": false,
      },
    }
  }
  className="undefined authors-publication-types-count-chart"
/>
`;
